packages/graphback/src/GraphQLBackend.ts

```diff
@@ -89,10 +94,24 @@ export class GraphQLBackendCreator {
       // logger.error(`Error on Database creation ${error}`)
       throw error
     }
+  }
 
+  /**
+   * Filters duplicate fields (by name) from a list of Fields
+   *
+   * @private
+   * @param {Field[]} fields The original list of fields from the type
+   * @returns {Field[]} Return the filtered list of fields
+   * @memberof GraphQLBackendCreator
+   */
+  private getUniqueFields(fields: Field[]): Field[] {
```

Comment:
minor nit: This is O(n^2) when it could be linear and it's really hard to understand. This would suffer performance degradation as the schema grows. There's nothing wrong with that but we might as well not let these perf things creep up on us.

There's a really nice stackoverflow answer that uses a set https://stackoverflow.com/questions/9229645/remove-duplicate-values-from-js-array

The code would look something like:

```
private getUniqueFields(fields: Field[]): Field[] {
    const seen = {};
    return fields.filter(function(field) {
		if (seen.hasOwnProperty(field.name)) {
			return false
		}
        seen[field.name] = true
		return true
    })
}
```

In this example we iterate over the array once using `filter`. For each item we check to see if `seen.hasOwnProperty(field.name)` is true. If it's not true, then then add it and return true. If `seen.hasOwnProperty(field.name)` is true that means we've hit a duplicate, so we return false, this ensures the duplicate is not in the newly returned array.

This code has the benefit of being faster (linear time) and in my opinion, a lot easier to understand.